nixos/modules/services/monitoring/pgscv.nix

```
@@ -0,0 +1,67 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  inherit (lib)
+    mkEnableOption
+    mkIf
+    mkOption
+    types
+    ;
+
+  cfg = config.services.pgscv;
+
+  settingsFormat = pkgs.formats.yaml { };
+  configFile = settingsFormat.generate "config.yaml" cfg.settings;
+in
+{
+  options.services.pgscv = {
+    enable = mkEnableOption "pgSCV, a PostgreSQL ecosystem metrics collector";
+
+    logLevel = mkOption {
+      type = types.enum [
+        "debug"
+        "info"
+        "warn"
+        "error"
+      ];
+      default = "info";
+      description = "Log level for pgSCV.";
+    };
+
+    settings = mkOption {
+      type = settingsFormat.type;
+      default = { };
+      description = ''
+        Configuration for pgSCV, in YAML format.
+
+        See [configuration reference](https://github.com/cherts/pgscv/wiki/Configuration-settings-reference).
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.pgscv = {
+      description = "pgSCV - PostgreSQL ecosystem metrics collector";
+      wantedBy = [ "multi-user.target" ];
+      requires = [ "network-online.target" ];
+      after = [ "network-online.target" ];
+      path = [ pkgs.glibc ]; # shells out to getconf
+
+      serviceConfig = {
+        User = "postgres";
+        Group = "postgres";
+        ExecStart = "${pkgs.pgscv}/bin/pgscv --log-level=${cfg.logLevel} --config-file=${configFile}";
```

```suggestion
        ExecStart = utils.escapeSystemdExecArgs ([
          (lib.getExe pkgs.pgscv)
          "--log-level=${cfg.logLevel}"
          "--config-file=${configFile}"
        ] ++ cfg.extraArgs);
```

Proper tokenization and escape hatch for for custom args.

---

These are the only two flags it has, applied the rest.